src/annotations.jl

```diff
@@ -30,6 +30,38 @@ Invoke/return `Legolas.write(path_or_io, annotations, Schema("onda.annotation@1"
 """
 write_annotations(path_or_io, annotations; kwargs...) = Legolas.write(path_or_io, annotations, Legolas.Schema("onda.annotation@1"); kwargs...)
 
+"""
+    validate_annotations(annotations)
+
+Perform both table-level and row-level validation checks on the content of `annotations`,
+a presumed `onda.annotation` table. Returns `annotations`.
+
+This function will throw an error in any of the following cases:
+
+- `Legolas.validate(annotations, Legolas.Schema("onda.annotation@1"))` throws an error
+- `Annotation(row)` errors for any `row in Tables.rows(annotations)`
+- `annotations` contains rows with duplicate `id`s
+"""
+function validate_annotations(annotations)
+    Legolas.validate(annotations, Legolas.Schema("onda.annotation@1"))
+    id_counts = Dict{UUID,Int}()
+    for (i, row) in enumerate(Tables.rows(annotations))
+        local annotation
+        try
+            annotation = Annotation(row)
+        catch err
+            log("Encountered invalid row $i of given `onda.annotation` table:")
+            rethrow(err)
+        end
+        id_counts[annotation.id] = get(id_counts, annotation.id, 0) + 1
+    end
+    filter!(((_, c),) -> c > 1, id_counts)
```

Comment:
not sure if this is any better

```suggestion
    filter!(>(1) ∘ last, id_counts)
```

Reply:
looks purdy